analyzers/src/SonarAnalyzer.CSharp/Rules/ExpressionComplexity.cs

```diff
@@ -46,7 +46,7 @@ public sealed class ExpressionComplexity : ExpressionComplexityBase<SyntaxKind>
         protected override SyntaxNode[] ExpressionChildren(SyntaxNode node) =>
             node switch
             {
-                ConditionalExpressionSyntax conditional => new[] { conditional.WhenTrue, conditional.WhenFalse },
+                ConditionalExpressionSyntax conditional => new[] { conditional.Condition, conditional.WhenTrue, conditional.WhenFalse },
```

Comment:
The condition expression was not visited. So we basically ignored all && and || in the conditional part (and sometimes didn't visit at all).